parquet-tools/src/main/java/org/apache/parquet/tools/read/SimpleMapRecord.java

```diff
@@ -30,14 +31,52 @@ protected Object toJsonObject() {
       String key = null;
       Object val = null;
       for (NameValue kv : ((SimpleRecord) value.getValue()).values) {
-        if (kv.getName().equals("key")) {
-          key = (String) kv.getValue();
-        } else if (kv.getName().equals("value")) {
-          val = toJsonValue(kv.getValue());
+        String kvName = kv.getName();
+        Object kvValue = kv.getValue();
+        if (kvName.equals("key")) {
+          key = keyToString(kvValue);
+        } else if (kvName.equals("value")) {
+          val = toJsonValue(kvValue);
         }
       }
       result.put(key, val);
     }
     return result;
   }
+
+  private String keyToString(Object kvValue) {
+    Class<?> type = kvValue.getClass();
+    if (type.isArray()) {
+
+      if (type.getComponentType() == boolean.class) {
+        return Arrays.toString((boolean[]) kvValue);
+      }
+      else if (type.getComponentType() == byte.class) {
+        return Arrays.toString((byte[]) kvValue);
```

Comment:
I think this should use the BinaryNode that is used elsewhere to convert a Binary or byte[] to JSON. That will base64-encode the data so it is still binary rather than creating an array node.